lib/theme/widgets/hs_warning_bottom_sheet.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:hatspace/dimens/hs_dimens.dart';
+import 'package:hatspace/theme/hs_theme.dart';
+import 'package:hatspace/theme/widgets/hs_buttons.dart';
+
+class HsWarningBottomSheetView extends StatelessWidget {
+  final String? iconUrl;
+  final String? title;
+  final String? description;
+  final String? primaryButtonLabel; // return false value
+  final VoidCallback? primaryOnPressed;
+  final String? secondaryButtonLabel; // return true value
+  final VoidCallback? secondaryOnPressed;
+  final String? textButtonLabel;
+  final VoidCallback? textButtonOnPressed;
+  final String? tertiaryButtonLabel;
+  final VoidCallback? tertiaryButtonOnPressed;
+  const HsWarningBottomSheetView({
+    super.key,
+    this.iconUrl,
+    this.title,
+    this.description,
+    this.primaryButtonLabel,
+    this.primaryOnPressed,
+    this.secondaryButtonLabel,
+    this.secondaryOnPressed,
+    this.textButtonLabel,
+    this.textButtonOnPressed,
+    this.tertiaryButtonLabel,
+    this.tertiaryButtonOnPressed,
+  })  : assert(
+            iconUrl != null ||
+                title != null ||
+                description != null ||
+                primaryButtonLabel != null ||
+                secondaryButtonLabel != null ||
+                textButtonLabel != null ||
+                tertiaryButtonLabel != null,
+            'Require at least one field to enable bottom sheet'),
+        assert(
+            primaryButtonLabel != null &&
+                (primaryOnPressed != null || primaryOnPressed == null),
```

Comment:
1. primaryButtonLabel != null && primaryOnPressed != null -> why do we need to assert this case?
2. primaryButtonLabel != null && primaryOnPressed == null -> should warning no onClick event for this button

Your text: 'Need button lable to enable button' is not match with your condition

Reply:
with your text, primaryButtonLabel == null will be correct